provider/nbd_image_export.py

```diff
@@ -102,6 +103,14 @@ def export_image(self):
     def stop_export(self):
         raise NotImplementedError()
 
+    def remove_image(self):
+        export_image = qemu_storage.QemuImg(
```

Comment:
Actually we also need to handle the customized-created image, see create_image, there is a create_image_cmd defined by user to create an image in a user predefined way, but we can also remove it by ourselves in py, anyway, the original design of nbd_image_export is not good, as it still 'can be used' for our testing, let's keep it.

Reply:
Another way is in clean_images, we can

self.get_image_by_tag(self.params["local_image_tag"]).remove()

in this way, we don't need to touch nbd_image_export, which one do you prefer @aliang123

Reply:
@zhencliu, thanks for your advice, will check it and update it later.

Reply:
@zhencliu, have updated, please help to review again, thanks.

Reply:
hi @aliang123 , the changes look good to me. 

But I suggest that it's better for us to use 2 commits to track these 2 issues: one is 'Remove local image after test' for py change, and the other is 'Skip expected warning msg of dmesg' for cfg change

why?
1. one day we want to revert this commit, both fixes will not be available, in fact, we may only want to revert one (This situation almost not happen though :-D, but this may happen in a product development)
2. Two different commits for two different issues, reasonable.

Reply:
@zhencliu , have splitted, you can check it again, thanks.